### PR TITLE
Refactor "highlight.js" Code for Improved Readability and Maintainabi…

### PR DESCRIPTION
**Pull request by TusharNaugain**
…lity

1 Broke down large functions into smaller, more manageable functions to improve readability and facilitate easier testing. Enhanced Comments and Documentation:

2. Added context and explanations in comments to help future developers understand the code better. Improved Error Handling:

3. Compiled frequently used regex patterns for better performance.

Use of Constants for Magic Strings:
4. Replaced hard-coded strings with constants to avoid typos and improve maintainability.

5. Removal of Deprecated Code: Eliminated deprecated functions and provided clear migration paths where necessary.

Performance Enhancements:
Reviewed loops and conditionals to ensure they are efficient and do not perform unnecessary operations.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by joshgoebel** [comment]
To even review this it would need to be MUCH smaller chunks, not remove all our types/comments, etc... perhaps if you want to pick out a few isolated functions and start there and submit a much smaller PR... if that goes well we could keep moving forward one bit at a time.

**Comment by joshgoebel** [comment]
I'll see if I can drop a few comments on things that look promising at first glance.

File: src/highlight.js [hunk 1]
```diff
@@ -74,667 +36,138 @@ const HLJS = function(hljs) {
...
-      modeBuffer += lexeme;
-      return lexeme.length;
-    }
-
     const language = getLanguage(languageName);
     if (!language) {
-      logger.error(LANGUAGE_NOT_FOUND.replace("{}", languageName));
-      throw new Error('Unknown language: "' + languageName + '"');
+      console.error(LANGUAGE_NOT_FOUND.replace("{}", languageName));
+      throw new Error(`Unknown language: "${languageName}"`);
```
Comment (joshgoebel) [inline src/highlight.js:74]: Little improvements like this (switching from + to `` interpolation) are nice.

**Comment by TusharNaugain** [comment]
> f I can drop a few comments on things that look promising at first glance.

okiee sir

**Comment by TusharNaugain** [comment]
> To even review this it would need to be MUCH smaller chunks, not remove all our types/comments, etc... perhaps if you want to pick out a few isolated functions and start there and submit a much smaller PR... if that goes well we could keep moving forward one bit at a time.

okiee sir i have clossed this pr now again i am raise a PR and then try this as per your guidance.

**Pull request closed without merging**